lib/wsfed.js

```diff
@@ -93,7 +93,7 @@ module.exports = function(options) {
         audiences:            audience,
         attributes:           claims,
         nameIdentifier:       ni.nameIdentifier,
-        nameIdentifierFormat: ni.nameIdentifierFormat,
+        nameIdentifierFormat: ni.nameIdentifierFormat || options.nameIdentifierFormat,
```

Comment:
Probably I'm missing execution context here, but, if there is a default value (according to tests), is that value on `ni.nameIdentifierFormat`? if so, I don't understand how the `ptions.nameIdentifierFormat` will take precedence when set.

Reply:
> if there is a default value (according to tests), is that value on `ni.nameIdentifierFormat`

Thanks for the comment, that is a good question and I wondered the same about where this default value was coming from, but no the default value is not on `ni.nameIdentifierFormat` from my understanding. The default value is set by the [SAML library](https://www.npmjs.com/package/saml) when the`nameIdentifierFormat` argument is undefined and in our case that would be when both `ni.nameIdentifierFormat` and `options.nameIdentifier` are undefined. 

`ni.nameIdentifierFormat` comes from a callback function passed to the `node-wsfed` library called the profile mapper, this value is assigned if that function returns this `nameIdentifierFormat` property.

So the order of precedence is as follows...
`ni.nameIdentifierFormat` > `options.nameIdenfifierFormat` > `saml nameIdentifierFormat (the default)`

Hopefully that make sense, but let me know if not. :)